### DISCUSSION TITLE
Is there a plan to support other snark-friendly hash function?

### DISCUSSION THREAD

**Discussion by jeljelih** [Consensys/gnark#904] [Ideas]
I know currently only MiMC is implemented in the standard library of gnark. Is there a plan to support other snark-friendly hash functions like Poseidon?

**Comment by gbotrel** [comment]
Hi -- I've seen some other projects implementing their own for example, Binance did Poseidon: https://github.com/bnb-chain/gnark/tree/zkbnb-dev/std/hash/poseidon .
